### PR TITLE
Add KubevirtClusterTemplate resource to support ClusterClass feature

### PR DESCRIPTION
**Pull request by aamoyel**
Fix: #250

### CODE REVIEW

**Comment by agradouski** [comment]
@davidvossel note, this is an alpha feature of CAPI, any objections to support it in CAPK at this time?

File: pkg/loadbalancer/loadbalancer.go
Comment (aamoyel) [inline pkg/loadbalancer/loadbalancer.go:1]: Like the Workload Cluster, we need to use the cluster name for LB metadata.name cause KubevirtCluster add a prefix on kubevirtCluster ressource when ClusterClass and KubevirtClusterTemplate is used.

File: api/v1alpha1/kubevirtclustertemplate_types.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=kubevirtclustertemplates,scope=Namespaced,categories=cluster-api,shortName=kct
+
+// KubevirtClusterTemplate is the Schema for the kubevirtclustertemplates API.
+type KubevirtClusterTemplate struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec KubevirtClusterTemplateSpec `json:"spec,omitempty"`
```
Comment (davidvossel) [inline api/v1alpha1/kubevirtclustertemplate_types.go:43]: could we add something like the validation below to ensure immutability without needing a webhook?
`        // +kubebuilder:validation:XValidation:rule="self == oldSelf", message="spec is immutable"`

**Review by davidvossel — COMMENTED** [review]
hey, I haven't looked deeply at the ClusterClass feature, but this api seems fine.

The only part I question is why we need a validating webhook to ensure this template remains immutable after creation. Can't we express immutability using the crd CEL validation?

File: api/v1alpha1/kubevirtclustertemplate_types.go (hunk 1 above)
Comment (aamoyel) [reply api/v1alpha1/kubevirtclustertemplate_types.go:43]: Hello David, thx for your review :) .
I didn't know it was possible to enforce CRD Immutability with CEL Rules, thanks for the advice. I will update my PR to match your specification.

**Comment by nunnatsa** [comment]
/hold

github action CI didn't run yet.

**Comment by nunnatsa** [comment]
/hold

The ci was not run yet

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Comment by nunnatsa** [comment]
> > github action CI didn't run yet.
> 
> oh? I see that the e2e integration test passed. What else are we waiting on?

Unit tests, linters, check-gen, etc.

Admind should see the "Approve and Test" button. Clicking on it will enable these tests

**Comment by nunnatsa** [comment]
> > > github action CI didn't run yet.
> > 
> > 
> > oh? I see that the e2e integration test passed. What else are we waiting on?
> 
> Unit tests, linters, check-gen, etc.
> 
> Admins should see the "Approve and Test" button. Clicking on it will enable these tests

@agradouski @cchengleo ^^^

**Comment by aamoyel** [comment]
Can you replay tests please @agradouski @davidvossel ?

**Pull request merged**